x/skyway/abci.go

```diff
@@ -42,6 +42,14 @@ func EndBlocker(ctx context.Context, k keeper.Keeper, cc *libcons.ConsensusCheck
 		if err != nil {
 			logger.WithError(err).Warn("Failed to prune attestations.")
 		}
+
+		// Update all validator nonces to the latest observed nonce, if suitable
+		// This makes sure validators don't get stuck waiting for events they
+		// can't get from the RPC, even though these events are already observed
+		err = k.UpdateValidatorNoncesToLatest(ctx, v)
+		if err != nil {
+			logger.WithError(err).Warn("Failed to update validator nonces.")
+		}
```

Comment:
Since this isn't mission critical, let's extract this into something that runs i.e. every 50 blocks or so to keep the block speed low.

Reply:
Thank. I added that condition and tested it in the private testnet.